hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -354,7 +365,9 @@ protected Expression getFilterPredicates(Configuration configuration) {
     DataMapJob dataMapJob = getDataMapJob(job.getConfiguration());
     List<PartitionSpec> partitionsToPrune = getPartitionsToPrune(job.getConfiguration());
     List<ExtendedBlocklet> prunedBlocklets;
-    if (distributedCG || dataMapExprWrapper.getDataMapType() == DataMapLevel.FG) {
+    if (isFgDataMapPruningEnable(job.getConfiguration()) &&
+        (distributedCG || dataMapExprWrapper.getDataMapType() == DataMapLevel.FG) &&
+        dataMapJob != null) {
```

Comment:
But I guess we should prune with blockletdatamap once if fg is disabled.

Reply:
I changed the condition again, please check